src/components/ActivityCard/ActivityCards.js

```diff
@@ -37,14 +38,35 @@ export default function ActivityCards({activities, onSetActivities, onSetIsHidde
     ]);
     closeModalHandler();
   };
-  //----- Show & Hide Input Modal -----
+  //----- handle Input Modal show and hide -----------------------------------------------------------------------------------------
+
   function showModalHandler() {
     setShowModal(true);
   }
 
   function closeModalHandler() {
     setShowModal(false);
   }
+  //----- handle delete Activities & confirmation Modal -----------------------------------------------------------------------------------------
+
+  const deleteActivity = () => {
+    if (showModalConfirmation.show && showModalConfirmation.id) {
+      const temp = activities.filter(acti => acti.id !== showModalConfirmation.id);
+      onSetActivities(temp);
```

Comment:
I would recommend against passing the `setState` function as props directly. Rather pass a callback function and have the functionality to update the state where the state is. For example, if you want to delete an activity, instead of filtering in your `ActivityCards` page, it should be done in the `App.js` where the state is. You then just pass a call back function back: `onDeleteActivity` which is called in the `App.js`. 

Passing the `setState` function directly to a component, might cause some bugs and weird behavior as updating the state is not done within the component where the state is but somewhere else.

Reply:
I tried this but i miss the id on the way and i'm not sure where

Reply:
Updated, thanks for the suggestions :)